Jatmika Teja

thejatmik opened the pull request:
Heroku
https://tranquil-lowlands-54278.herokuapp.com/

thejatmik committed:
Ecommerce Server

Initialize server, testing purpose

thejatmik committed:
Register and Login Testing

thejatmik committed:
Initialize product testing

thejatmik committed:
Product fetch testing

dhimashary commented on middlewares/authenticator.js:
```diff
@@ -0,0 +1,30 @@
+module.exports = function (req, res, next) {
+    const access_token = req.headers.access_token;
+    if (!access_token) {
+        next(appError(401, "please login as valid user"));
```
next itu nggak memberhentikan function di middleware ya

dhimashary replied:
jadi baiknya dikasih return juga, buat mastikan statement dibawah ini pasti gak ke run

dhimashary commented on middlewares/authenticator.js:
```diff
@@ -0,0 +1,30 @@
+    if (!payload.email || !payload.id) {
+        next(appError(401, "appPayload not found"));
+    }
+    User.findOne({
```
User ini maksudnya admin ya?

thejatmik replied:
iya mas, utk sekarang semua user masih role admin. belum ada pembagian role

dhimashary commented on test/user.test.js:
```diff
@@ -0,0 +1,190 @@
+                        expect(err).toBeNull();
+                        expect(res.body).toHaveProperty('message', 'bad request');
+                        expect(res.body).toHaveProperty('errors', expect.any(Array));
+                        expect(res.body.errors.length).toBeGreaterThan(0);
```
mending langsung di cek aja .toBe(1)

dhimashary commented on test/product.test.js:
```diff
@@ -0,0 +1,296 @@
+
+beforeAll((done) => {
+    User.create(user1)
+    .then(result => {
```
then catch promise tab sekali ya indent nya

dhimashary commented on test/product.test.js:
```diff
@@ -0,0 +1,296 @@
+            })
+        });
+    });
+    // describe('user add product', () => {
```
kok yang bawahnya ini di komen ya

thejatmik replied:
rute post /product belum dibuat, jadi masih belum diaktifkan & masih dipake panduan utk buat rutenya...

thejatmik committed:
Product testing and server fix

- fix authenticator
- fix semantics
- User testing, narrowing expectation
- User model, add unique constraint
- Product testing for fetch and create new item

thejatmik committed:
Some issues with client interaction

- Not receiving access token, fixed
- Cors, fixed
- get /product still limited to fetch owned product

thejatmik committed:
Product Route Finished

- GET, POST, PUT, and DELETE
- Tested with jest

thejatmik committed:
Finishing server v01

- Product and User routes tested
- Product CRUD
- User login and register

thejatmik committed:
Deployment 01

thejatmik committed:
fix deployment 01 issues

dhimashary commented on middlewares/authenticator.js:
```diff
@@ -0,0 +1,33 @@
+                })
+                    .then(result => {
+                        if (!result) {
+                            next(appError(401, "please login as valid user"));
```
kok gak nyimpen current user yang lagi login disini ya?

dhimashary commented on controllers/ProductController.js:
```diff
@@ -0,0 +1,113 @@
+        .catch(next);
+    }
+    static create(req, res, next) {
+        const user = appPayload(getPayload(req.headers.access_token));
```
kalo current user yang lagi login disimpan pas lewat middleware auth ini ga perlu dibikin lagi

dhimashary replied:
tolong dibenahi sebelum jumat ya

dhimashary replied:
EH tapi kan ga perlu ada UserId ya di produk karena semua admin boleh create, dan ga ada relasi

dhimashary commented on models/product.js:
```diff
@@ -0,0 +1,92 @@
+        },
+      }
+    },
+    UserId: {
```
ga perlu ada UserId di product!

dhimashary commented on models/product.js:
```diff
@@ -0,0 +1,92 @@
+    modelName: "Product",
+    hooks: {
+      beforeValidate: (product, options) => {
+        if (!product.image_url) {
```
jangan taro di hooks, dari awal pas add udah harus masukin gambar

thejatmik committed:
Init Customer development

thejatmik committed:
Customer Login and Register testing

thejatmik committed:
Customer fetch product

- Init customer fetch route
- Fetch route tested

thejatmik committed:
Customer fetch cart

- Init route and testing

thejatmik committed:
Customer add item to cart

- route initialized and tested

thejatmik committed:
Cart Item delete

- init route and tested

thejatmik committed:
Cart Item update

- Init route and tested

thejatmik committed:
Server add feature

- Clear customer cart
- tested on success

thejatmik committed:
Deployment

- add config/
- modify .gitignore

thejatmik committed:
Refactor Server

thejatmik committed:
Admin Documentation

thejatmik committed:
Customer Register and Login Documentation

thejatmik committed:
customer documentation

dhimashary commented on controllers/CustomerController.js:
```diff
@@ -0,0 +1,383 @@
+      .then(result => {
+        // console.log(result)
+        let cartItems = []
+        result.forEach(item => {
```
kenapa perlu di loop lagi y?
